linera-storage/src/lib.rs

```diff
@@ -115,9 +115,12 @@ pub trait Storage: Sized {
         hash: CryptoHash,
     ) -> Result<HashedCertificateValue, ViewError>;
 
-    /// Reads the blob with the given blob ID.
+    /// Reads the hashed blob with the given blob ID.
     async fn read_hashed_blob(&self, blob_id: BlobId) -> Result<HashedBlob, ViewError>;
 
+    /// Reads the blob with the given blob ID.
+    async fn read_blob(&self, blob_id: BlobId) -> Result<Blob, ViewError>;
```

Comment:
That feels redundant: We can just use `read_hashed_blob` and throw away the ID where we don't need it. It's unchecked, so we're not computing an expensive hash anyway.